### PR TITLE
fix mypy on `VisionDataset`

### PR DESCRIPTION
**Pull request by pmeier**
Supersedes #8128. Error caused by #8124.

### CODE REVIEW

File: torchvision/datasets/fakedata.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def __init__(
         target_transform: Optional[Callable] = None,
         random_offset: int = 0,
     ) -> None:
-        super().__init__(None, transform=transform, target_transform=target_transform)  # type: ignore[arg-type]
+        super().__init__(transform=transform, target_transform=target_transform)
```
Comment (pmeier) [inline torchvision/datasets/fakedata.py:34]: Driveby, since we no longer need to pass `None` after #8124.

File: torchvision/datasets/vision.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ class VisionDataset(data.Dataset):
 
     def __init__(
         self,
-        root: Optional[str] = None,
+        root: str = None,  # type: ignore[assignment]
```
Comment (NicolasHug) [inline torchvision/datasets/vision.py:32]: can we still keep the `Optional[str]` type which is technically correct?

**Review by NicolasHug — APPROVED** [review]
Thanks Philip. One Q but LGTM anyway

File: torchvision/datasets/vision.py (hunk 2 above)
Comment (pmeier) [reply torchvision/datasets/vision.py:32]: We can't. If we use `Optional[str]` as annotation, `mypy` correctly infers that for the `self.root` attribute. And with that we are back at the issues we are currently seeing, i.e. all child datasets that perform anything with `self.root` would now need to check that `self.root` is in fact a `str` and not `None`.
